### ISSUE TITLE
Data from `file.data` not overriding `app.cache.data`

### ISSUE THREAD

**Issue by dtothefp** [jonschlinkert/templates#19]
I thought in the past if I did 

``` js
app.data({bleep: 'bleep'})
app.onLoad(/\.html$/, (file, next) => {
  file.data = {bleep: 'bloop'}
  next(null, file);
})
```

that the same key on `file.data` would overwrite that for `app.cache.data` but that doesn't seem to be the case

here's an example, run `node index.js`

https://github.com/dtothefp/assemble-view-issue

**Comment by doowb** [comment]
I think I've tracked down where this can be fixed. I'll put a PR together for discussion and to determine if changes will cause breaking changes downstream.

**Comment by jonschlinkert** [comment]
@doowb was this resolved?

**Comment by dtothefp** [comment]
I think @doowb said this was resolved in assemble-core 0.16.0 but I haven't tested yet

**Comment by doowb** [comment]
Yes. This was resolved with the context method updates. Closing but feel free to reopen if there's still an issue.
